src/components/searchSection/catergrayList.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { MDBIcon } from 'mdbreact';
+import './myStyle.scss';
+
+function CatergrayList({ items }) {
+  let itemsTags = [];
+  let numberOfItems = items.length;
+  let [ActiveBubble, setActiveBubble] = useState(0);
+  let [setCatogray] = useState('');
+
+  //light the bubble by pink color
+  const handelBubble = (index) => {
+    if (ActiveBubble === index) {
+      return 'textPink';
+    }
+  };
+
+  const emptyItems = () => {
+    for (let i = 1; i <= 9 - numberOfItems; i++) {
+      itemsTags.push(<div md="12" className="lightBlack h-10 mb-1"></div>);
+    }
+  };
+  const getItemList = (e) => {
+    if (e.target !== undefined) {
+      setCatogray(e.target.children[1].innerHTML);
+    }
+  };
+
+  const typeItems = () => {
+    itemsTags = items.map((item, index) => {
+      return (
+        <div
+          className="text-left lightBlack h-10 mb-1 "
+          key={index}
+          //turn the light to pink for the item that choose and the other to unactive
+          onClick={(e) => {
+            setActiveBubble(index);
+            getItemList(e);
+          }}
+        >
+          <MDBIcon
+            icon="circle"
+            className={`p-3 sizeBubble ${handelBubble(index)}`}
+          />
+          <p className="parg1 d-inline hoverCursor">{item.name}</p>
+        </div>
+      );
+    });
+  };
+
+  typeItems();
+  emptyItems();
+  return itemsTags;
+}
```

Comment:
this logic is a big mess
- instead of storing the map result in a variable and then use the variable you can return the result of map and use the function directly to render the items
- having a function that do a side effect is always a bad practice search about that
- the last function on line 52 is unclear 
- even the function `handlBubble` on line 12 is a bit confusing, instead call it `isActive` and this function will return true or false and depending on this return you will decide to put the class name or not

contact me if you need more help here